### ISSUE TITLE
instagramLink blueprint

### ISSUE THREAD

**Issue by thewebprojects** [genxbe/kirby3-instagram#4]
Hoping you can help me through some setup of this plugin (which I greatly appreciate you taking the time to build).  With so many hoops to jump though I'd like to confirm a few things...

Your instructions state, "find a place to add the instagramLink blueprint".  What does that mean? Is this just referring to me adding "linkInstagram" to one of my page blueprints.  For example, I want to feature a Instagram feed on my home page so I'll add this to my home.yml file.  Please confirm.

**Comment by samzzi** [comment]
The easiest is to add the linkInstagram to your site blueprint.
I do it on a settings tab on the `site.yml` most of the times but obviously you're free to choose where you want to put it.

In the example in the readme it's also added on `site.yml`

```yml
title: Site
preset: pages
unlisted: true

fields:
  linkInstagram: linkInstagram
```

**Comment by thewebprojects** [comment]
Gotcha. I did put it on my home page blueprint.  What does "Instagram link: **** (0 is inactive, 1 is active)" mean? 

I didn't know this plugin would create the link the Instagram owner/client had to click on so I generated one myself and she did visit the link and allow. Does "1 active" mean it's setup correctly?  Because I'm not seein' the Instagram feed on my home page but that could be another issue.  Please advise.

**Comment by samzzi** [comment]
If it's active it's ok. Next step is to setup a cron, as long as the command is not executed no pictures are downloaded. You can also run it manually but in the end the cron needs to be setup because the instagram token needs to be renewed now and then.

If you want to test it manually you can execute `php site/plugins/kirby3-instagram/fetch.php` in the root of your site. Don't do that to much though or instagram will be annoyed (there are limits, that's why the cron is only executed every 30 minutes)

**Comment by thewebprojects** [comment]
I see.  Okay so until that fetch.php is run/executed nothing will work.  I can imagine how we'd run that with a cron (and I will set that up), but I wanna just confirm that it'll work now.  How would I run that script?  There's no URL to that file.  Please advise.
And thank you for your quick replies.  Much appreciated.

**Comment by samzzi** [comment]
Open up a terminal in the root folder (for example `/home/user/websites/mysite.com/`) of your site and run `php site/plugins/kirby3-instagram/fetch.php`

Or if you're working on a server directly you will need to use SSH.

**Comment by thewebprojects** [comment]
Oh. Right, the geeky way via SSH. Ha.  Got it.  Thanks @samzzi. That's very helpful.

**Comment by thewebprojects** [comment]
Oh lord. I just asked the client web host, iPower, where the heck I can setup cron jobs. The don't have the ability to setup cron jobs.  Wow.  Is there a way to make that fetch.php file available via a url? Then I can setup some monitoring service to run it every 30 mins or so.

**Comment by samzzi** [comment]
Well, I think you could add a route that executes the fetch command.

Something like (untested -- if you don't use composer it's possible that you will have to include/autoload the class first)

### /site/config/config.php
```php
return [
  'routes' => [
    [
      'pattern' => 'cron/fetch/instagram/somerandomunguessablehash',
      'action'  => function () {
        if(site()->instagramToken()->IsEmpty())
        {
            throw new \Exception('Instagram token is required.');
        }

        (new X\Instagram)->fetch();
      }
    ],
  ]
];
```

**Comment by thewebprojects** [comment]
Thanks for the suggestion.  I opted to create a route that requires the fetch.php file.

```
'routes' => [
    [
      'pattern' => 'fetch-instagram-feed',
      'action'  => function () {
        require(__DIR__.'/../plugins/kirby3-instagram/fetch.php');
      }
    ]
  ]
```
Upon visiting that URL it throws the 'Instagram token is required.' Could that mean the token expired.  I suppose it's be hours now since the client originally clicked on the generated link (which prompts them to allow...).  The field in the panel still states, "Instagram link: **** (0 is inactive, 1 is active)"

![ss](https://user-images.githubusercontent.com/19271269/163024311-c92a1e92-8324-48c5-843d-ef81ff38de7e.jpg)

Any insight? I know I'm close to gettin' this up and running. 
Thanks for your time @samzzi!

**Comment by samzzi** [comment]
I don't think the instagram is connected with the account. It should say 1 and be green

https://x.gnx.li/x2AcXV

Can't you test with your own account to make sure everything works? Maybe the client did something wrong with the connection? (shouldn't be that hard though, they only need to click allow)

Did you add the account to the test users in your instagram app?

**Comment by thewebprojects** [comment]
Hey @samzzi, Yeah. The test user is setup in the app.  

![ss](https://user-images.githubusercontent.com/19271269/163042335-22d01aec-7f0b-4bb1-88e6-ac357e690a9a.jpg)

Hmmm. I'm wondering if that route in Kirby isn't working as expected? Maybe because I'm requiring the fetch.php file from the config directory the following (which lives in the fetch.php file) isn't loading.  
```
$files = [
    __DIR__ . '/vendor/autoload.php',
    realpath(__DIR__ . '/../../../') . '/vendor/autoload.php',
    __DIR__ . '/kirby/bootstrap.php',
    realpath(__DIR__ . '/../../../') . '/kirby/bootstrap.php',
];
```
I suppose if there's any error in that fetch.php file it'll state, "Instagram token is required.".  So maybe it's not the token, but the fact that something else isn't correct. Hmm.

**Comment by samzzi** [comment]
You can easily check if the token is there by checking your site.txt

You should find the following entries in the site.txt

```
----

Instagramauth: 1

----

Instagramtoken: IGQVJXd.......GbGNR

----

Instagramtokenrefreshed: 2022-03-07
```

**Comment by thewebprojects** [comment]
I don't see the token on my home.txt content file.  

GOT IT TO WORK!

Instead of inserting `linkInstagram: linkInstagram` on my homepage blueprint I put it in my site blueprint. I clicked the link and allowed (my client gave me access to their Instagram account so I can do it).  I could then see the token data on the site.txt file. 

I visited the custom route and didn't get the error this time.  

I can confirm that it now works as I see the client's Instagram feed on my home page.  

So I wonder if somewhere in your code your assuming the user will place the `linkInstagram: linkInstagram` on the site blueprint?  

Anyway... thanks for your help!
